File: ios/VideoApp/Sources/Models/WatchHistory.swift
import Foundation

struct WatchHistoryItem: Decodable, Identifiable {
    let videoId: String
    let title: String?
    let coverUrl: String?
    let watchTime: String?

    var id: String { videoId + (watchTime ?? "") }
}

struct WatchHistoryResponse: Decodable {
    let list: [WatchHistoryItem]
    let total: Int
    let page: Int
    let pageSize: Int
}

File: android/app/src/main/java/com/github/makewheels/video2022/data/model/WatchHistory.kt
package com.github.makewheels.video2022.data.model

data class WatchHistoryItem(
    val videoId: String,
    val title: String?,
    val coverUrl: String?,
    val watchTime: String?
)

data class WatchHistoryResponse(
    val list: List<WatchHistoryItem>,
    val total: Long,
    val page: Int,
    val pageSize: Int
)